snapshot/lib/assets/SnapshotHelper.swift

```diff
@@ -32,9 +32,10 @@ func snapshot(_ name: String, waitForLoadingIndicator: Bool) {
 /// - Parameters:
 ///   - name: The name of the snapshot
 ///   - timeout: Amount of seconds to wait until the network loading indicator disappears. Pass `0` if you don't want to wait.
+///   - screenshotProvider: The provider to screenshot. Otherwise `XCUIScreen.main` is screenshotted. This is useful when taking screenshots for macOS and you'd prefer to screenshot the window and not the screen.
 @MainActor
-func snapshot(_ name: String, timeWaitingForIdle timeout: TimeInterval = 20) {
-    Snapshot.snapshot(name, timeWaitingForIdle: timeout)
+func snapshot(_ name: String, timeWaitingForIdle timeout: TimeInterval = 20, screenshotProvider: XCUIScreenshotProviding = XCUIScreen.main) {
+    Snapshot.snapshot(name, timeWaitingForIdle: timeout, screenshotProvider: screenshotProvider)
```

Comment:
The XCUIScreen.main gives me a Swift6 bug. You might need to look into it.

"Main actor-isolated class property 'main' can not be referenced from a nonisolated context; this is an error in the Swift 6 language mode"